app/src/components/page/Sidebar/utils.tsx

```diff
@@ -0,0 +1,23 @@
+import { Route } from '../../../routes';
+
+/**
+ *
+ * @param routes - The routes to get search the active route from
+ * @param path - an array of menu location path
+ */
+export function getActiveKey(path: string, routes: Route[]) {
+  let activeKey: string | undefined;
+
+  function mapMenus(menu: Route) {
+    // Matching Url
+    if (menu.url && path.includes(menu.url)) activeKey = menu.key;
+
+    // Exact Match
+    if (path === menu.url) activeKey = menu.key;
+    // Trying to Match with Children
```

Comment:
@ashfahan again, do we need both of these 2 conditions?

Reply:
Yes we do need two condition because 
- if the relative Matching url is matched we still need to check if the current one exactly matched the url
- the else condition should be dependent on the exactly matching url because even if the first matching condition is matched what if the exactly matching url inside one of its children so the relative matching url should be a separate condition not conflicting or dependent on Exatcly matching url

Reply:
path.includes(menu.url) also does full string equality check. I.e path.includes(menu.url) will also return true for path === menu.url. The first check only should suffice

Reply:
if we remove the 2nd if condition and run the recursive function with out any check (just so i can check for the exactly match url in child) in that case it will keep on running even if we have already found one. and might be overridden by some other routes that at least matches this url thus its right to put a check and stop the recursion rather then letting it proceed forward